### PR TITLE
Add/edit/view Goods sections

### PR DESCRIPTION
**Pull request by kevinetienne**
Add forms and views for the following goods:
 - section 1;
 - section 2;
 - section 5;
 - section 58(2) - obsolete;
 - section 58(2) - other.

### CODE REVIEW

File: web/domains/case/_import/fa_sil/urls.py [hunk 1]
```diff
@@ -1,8 +1,76 @@
-from django.urls import path
+from django.urls import include, path
 
 from . import views
 
 app_name = "fa-sil"
 urlpatterns = [
     path("<int:pk>/edit/", views.edit, name="edit"),
+    # Goods
+    path("<int:pk>/sections/choice/", views.choose_goods_section, name="choose-goods-section"),
```
Comment (oskusalerma) [inline web/domains/case/_import/fa_sil/urls.py:9]: s/choice/choose/

File: web/domains/case/_import/fa_sil/views.py [hunk 2]
```diff
@@ -40,3 +53,252 @@ def edit(request: HttpRequest, pk: int) -> HttpResponse:
         }
 
         return render(request, "web/domains/case/import/fa-sil/edit.html", context)
+
+
+@login_required
+@permission_required("web.importer_access", raise_exception=True)
+def choose_goods_section(request: HttpRequest, *, pk: int) -> HttpResponse:
+    with transaction.atomic():
+        application = get_object_or_404(models.SILApplication.objects.select_for_update(), pk=pk)
```
Comment (oskusalerma) [inline web/domains/case/_import/fa_sil/views.py:62]: can type this variable, same in all other funcs

**Review by oskusalerma — CHANGES_REQUESTED** [review]

File: web/domains/case/_import/fa_sil/views.py (hunk 2 above)
Comment (kevinetienne) [reply web/domains/case/_import/fa_sil/views.py]: fixed in https://github.com/uktrade/icms/pull/383/commits/7ca781037bff7310a53bf45e85e6d89d6a57cb9a

File: web/domains/case/_import/fa_sil/urls.py [hunk 3]
```diff
@@ -1,8 +1,20 @@
...
 app_name = "fa-sil"
 urlpatterns = [
     path("<int:pk>/edit/", views.edit, name="edit"),
+    # Goods
+    path("<int:pk>/sections/choose/", views.choose_goods_section, name="choose-goods-section"),
+    path(
+        "<int:application_pk>/<silsectiontype:sil_section_type>/",
+        include(
+            [
+                path("add/", views.add_goods, name="add-section"),
```
Comment (oskusalerma) [inline web/domains/case/_import/fa_sil/urls.py:14]: why the mismatch between function/view names? they should both be goods or section, confusing now when they're different. goes for all of these.

File: web/domains/case/_import/fa_sil/views.py [hunk 4]
```diff
@@ -40,3 +61,169 @@ def edit(request: HttpRequest, pk: int) -> HttpResponse:
...
+            "process_template": "web/domains/case/import/partials/process.html",
+            "process": application,
+            "task": task,
+            "page_title": "Firearms and Ammunition (Specific Import Licence) - Edit Goods",
+        }
+
+        return render(request, "web/domains/case/import/fa-sil/choose-goods-section.html", context)
+
+
+class CreateSILSectionConfig(NamedTuple):
```
Comment (oskusalerma) [inline web/domains/case/_import/fa_sil/views.py:89]: please have all type definitions at top of files, not hidden between functions

File: web/domains/case/_import/fa_sil/views.py [hunk 5]
```diff
@@ -40,3 +61,169 @@ def edit(request: HttpRequest, pk: int) -> HttpResponse:
...
+            template="web/domains/case/import/fa-sil/goods/section582-obsolete.html",
+        )
+
+    elif sil_section_type == "section582-other":
+        return CreateSILSectionConfig(
+            model_class=models.SILGoodsSection582Other,
+            form_class=forms.SILGoodsSection582OtherForm,
+            template="web/domains/case/import/fa-sil/goods/section582-other.html",
+        )
+    raise NotImplementedError("sil_section_type is not supported: {sil_section_type}")
```
Comment (oskusalerma) [inline web/domains/case/_import/fa_sil/views.py:130]: you forgot f prefix

**Review by oskusalerma — CHANGES_REQUESTED** [review]

File: web/domains/case/_import/fa_sil/views.py (hunk 5 above)
Comment (kevinetienne) [reply web/domains/case/_import/fa_sil/views.py]: updated in https://github.com/uktrade/icms/pull/383/commits/2dca9e9eb3efc4243607770a1f18558fbcfdbf08

File: web/domains/case/_import/fa_sil/views.py (hunk 4 above)
Comment (kevinetienne) [reply web/domains/case/_import/fa_sil/views.py]: Updated in https://github.com/uktrade/icms/pull/383/commits/2dca9e9eb3efc4243607770a1f18558fbcfdbf08

File: web/domains/case/_import/fa_sil/urls.py (hunk 3 above)
Comment (kevinetienne) [reply web/domains/case/_import/fa_sil/urls.py]: Yeah it shows I was hesitating between goods and sections, fixed in https://github.com/uktrade/icms/pull/383/commits/2dca9e9eb3efc4243607770a1f18558fbcfdbf08

**Review by oskusalerma — APPROVED** [review]

**Pull request merged**